lib/elixir/lib/module.ex

```diff
@@ -861,6 +861,44 @@ defmodule Module do
     end, tuples)
   end
 
+  @spec make_overridable(module, module) :: :ok
+  def make_overridable(module, behaviour) when is_atom(module) and is_atom(behaviour) do
+    case check_module_for_overridable(module, behaviour) do
+      :ok -> :ok
+      {:error, error_explanation} ->
+        raise ArgumentError, "cannot pass module #{inspect(behaviour)} as argument to defoverridable/1 because #{error_explanation}"
+    end
+
+    behaviour_callbacks = :lists.map(&normalize_macro_or_function_callback/1, behaviour.behaviour_info(:callbacks))
+
+    tuples = for function_tuple <- definitions_in(module),
+                 function_tuple in behaviour_callbacks,
+             do: function_tuple
+
+    make_overridable(module, tuples)
+  end
+
+  @spec check_module_for_overridable(module, module) :: String.t
+  defp check_module_for_overridable(module, behaviour) do
+    behaviour_definitions = :ets.lookup_element(data_table_for(module), :behaviour, 2)
+
+    cond do
+      !Code.ensure_compiled?(behaviour)                  -> {:error, "it was not defined"}
+      !function_exported?(behaviour, :behaviour_info, 1) -> {:error, "it does not define any callbacks"}
+      !:lists.member(behaviour, behaviour_definitions)   -> {:error, "its corresponding @behaviour #{inspect(behaviour)} was not specified"}
```

Comment:
I'd like to see 2. implemented. Since 1. feels like we're asking for action if we already know what needs to be done.
Exactly the same way we have with `use` – never ask for `require` before, but do the call automatically.